rust/src/action.rs

```diff
@@ -54,6 +55,13 @@ fn gen_action_type_error(action: &str, field: &str, expected_type: &str) -> Acti
     ))
 }
 
+fn decode_path(raw_path: &str) -> String {
+    percent_decode(raw_path.as_bytes())
+        .decode_utf8()
+        .unwrap()
```

Comment:
I think it would be better to return a result in this function instead of a force unwrap.

Reply:
Sure, I wrapped it with a possible ActionError.